proposals/0002-rest-spread-operator.md

```diff
@@ -0,0 +1,133 @@
+# Feature name
+
+* Proposal: [HXP-0002](0002-rest-spread-operator.md)
+* Author: [Thomas Fétiveau](https://github.com/zabojad)
+
+
+## Introduction
+
+Add ES6-like `...` spread and rest operator to the Haxe syntax on both [arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator) and [objects](https://babeljs.io/docs/plugins/transform-object-rest-spread/).
+
+
+## Motivation
+
+Currently, there is a gap in Haxe for object duplication and overriding:
+
+* A lot of very popular js libs like React or Redux require developers to implement [pure functions](https://en.wikipedia.org/wiki/Pure_function) massively and this operator makes it really easy to do and avoid over-verbosity of simple operations.
+  
+  This:
+  ```
+  function pure(data : MyStruct) {
+
+  	return {...data, overwrittenFieldName: overwrittenFieldValue};
+  }
+  ```
+
+  is much less verbose than this:
```

Comment:
It's not just verbose, I think it's also worse performance, wise. If for `...obj, field2: somevalue` we generate an object decl like `{field1: obj.field1, field2: somevalue}`, the output will be cleaner and more performant.